src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TypeEnter.java

```diff
@@ -356,8 +356,10 @@ private void resolveImports(JCCompilationUnit tree, Env<AttrContext> env) {
 
                 // Import-on-demand java.lang.
                 PackageSymbol javaLang = syms.enterPackage(syms.java_base, names.java_lang);
-                if (javaLang.members().isEmpty() && !javaLang.exists())
-                    throw new FatalError(diags.fragment(Fragments.FatalErrNoJavaLang));
+                if (javaLang.members().isEmpty() && !javaLang.exists()) {
+                    log.error(Errors.NoJavaLang);
+                    throw new Abort();
+                }
```

Comment:
`Abort` seems to have different semantics to `FatalError`.
Unless there is a good reason for the change, it might be more advisable to stay with `FatalError` and just change the text of the message.

Reply:
"no java.lang" seems similar to a completion error, which is normally handled with `Abort`, so that might be a good reason for the change.